src/stream.rs

```diff
@@ -185,6 +188,27 @@ impl CpalDeviceExt for cpal::Device {
         .map(|stream| (mixer_tx, stream))
     }
 
+    fn try_new_output_stream(
+        &self,
+    ) -> Result<(Arc<DynamicMixerController<f32>>, cpal::Stream), StreamError> {
+        // Determine the format to use for the new stream.
+        let default_format = match self.default_output_config() {
+            Ok(f) => f,
+            Err(_) => return Err(StreamError::NoDevice),
+        };
+
+        Ok(self
+            .new_output_stream_with_format(default_format)
+            .unwrap_or_else(|err| {
+                // look through all supported formats to see if another works
+                supported_output_formats(self)
+                    .filter_map(|format| self.new_output_stream_with_format(format).ok())
+                    .next()
+                    .ok_or(err)
+                    .expect("build_output_stream failed with all supported formats")
```

Comment:
Is there a way to convert this into an Err as well?

Reply:
@est31  prob, but i am not sure if this should be a NoDevice or another or a new error in the enum.
As i have no big domain knowledge i just did the minimal change which i could test (by deleting pulsaudio-alsa package) to reproduce that this is catched.
Feel free to put a commit on top :) but as ve have many people complaining why rodio is panicing we wanted to get a fix quick for now :)